x-pack/platform/plugins/shared/alerting/server/routes/rule/apis/disable/disable_rule_route.ts

```diff
@@ -57,14 +57,33 @@ export const disableRuleRoute = (
     },
     router.handleLegacyErrors(
       verifyAccessAndContext(licenseState, async function (context, req, res) {
-        const rulesClient = await (await context.alerting).getRulesClient();
+        const alertingContext = await context.alerting;
+        const rulesClient = await alertingContext.getRulesClient();
+        const ruleTypes = alertingContext.listTypes();
+
         const { id }: DisableRuleRequestParamsV1 = req.params;
         const body: DisableRuleRequestBodyV1 = req.body || {};
         const { untrack = false } = body;
 
         const disableParams = { id, untrack };
 
         try {
+          const rule = await rulesClient.get({ id });
```

Comment:
Why don't these(disable+enable) use `validateInternalRuleTypesByQuery`? Is it because of  `getRuleTypesByQuery`? The performance should be fine if `filter` is optional and only an `id` is passed.

Reply:
The `validateInternalRuleTypesByQuery` does an ES aggregation to see if the rule is internal. I find it simpler to reason, explicit, and more performant if we do a single check. Also, if in the future the `validateInternalRuleTypesByQuery` changes to a way that supports only bulk edits, we will not have to go to all single routes and change the logic.